velox/core/SimpleFunctionMetadata.h

```diff
@@ -394,6 +394,46 @@ class ISimpleFunctionMetadata {
   virtual bool physicalSignatureEquals(
       const ISimpleFunctionMetadata& other) const = 0;
   virtual std::string helpMessage(const std::string& name) const = 0;
+
+  /// Returns string of all the fields such as logical signature,
+  /// physical signature and priority.
+  std::string toDebugString() const {
+    auto logicalArguments =
+        argumentToString<exec::TypeSignature>(signature()->argumentTypes());
+    auto physicalArguments = argumentToString<TypePtr>(argPhysicalTypes());
+
+    return fmt::format(
+        "Logical signature: ({}) -> {}\nPhysical signature: ({}) -> {}\n"
+        "Priority: {}\nDefaultNullBehavior: {}",
+        logicalArguments,
+        signature()->returnType().toString(),
+        physicalArguments,
+        resultPhysicalType()->toString(),
+        priority(),
+        defaultNullBehavior());
+  }
+
+ private:
+  template <typename T>
+  std::string argumentToString(const std::vector<T>& arguments) const {
+    std::stringstream ss;
+    bool first = true;
+    for (const auto& arg : arguments) {
+      if (!first) {
+        ss << ", ";
+      }
+      first = false;
+      if constexpr (std::is_same_v<T, exec::TypeSignature>) {
+        ss << arg.toString();
+      } else {
+        ss << arg->toString();
+      }
+    }
+    if (signature()->variableArity()) {
```

Comment:
It would be cleaner to pass this boolean as an argument. Otherwise, it is a bit strange that argumentToString takes args as input, but figures out whether args are variadic or not some other way.